File: sdks/rust/src/resources/evaluations.rs
use crate::client::BigRag;
use crate::error::BigRagError;
use crate::types::evaluations::{EvalBody, EvalResponse};

/// Evaluations resource.
pub struct Evaluations<'a> {
    pub(crate) client: &'a BigRag,
}

impl Evaluations<'_> {
    /// Run a retrieval evaluation.
    pub async fn run(&self, body: EvalBody) -> Result<EvalResponse, BigRagError> {
        self.client.transport.post("/v1/evaluation", &body).await
    }
}

File: sdks/typescript/src/resources/evaluations.ts
import type { RequestClient } from "../core.js";
import type { EvalBody, EvalResponse } from "../types/index.js";

export class EvaluationsResource {
  constructor(private readonly _client: RequestClient) {}

  run(body: EvalBody): Promise<EvalResponse> {
    return this._client._request("POST", "/v1/evaluation", { json: body });
  }
}
